src/editors/containers/ProblemEditor/data/ReactStateOLXParser.js

```diff
@@ -96,42 +97,91 @@ class ReactStateOLXParser {
     return compoundhint;
   }
 
-  addQuestion() {
+  parseQuestionFromState() {
     const { question } = this.problemState;
-    const questionObject = this.parser.parse(question);
+    const questionObject = new XMLParser({
+        ...this.parserOptions,
+        preserveOrder: true,
+        trimValues: false,
+      }).parse(question);
     return questionObject;
   }
 
+  insertQuestionIntoOLX(OLX, problemType, indentationLevel) {
+    // it's necessary to build the question with the different options, but
+    // this options will mess with the rest of the generated OLX, so we build
+    // them separately, and merge two strings; hacky, but works nicely without
+    // messing up formatting of the question or the rest of the XML
+    const question = this.parseQuestionFromState();
+    const questionBuild = new XMLBuilder({
+      ...this.builderOptions,
+      preserveOrder: true,
+      format: false,
+    }).build(question);
+
+    const tabWidth = 2;
+    const tagPattern = new RegExp(`<${problemType}[^>]*>`);
+
+    const match = OLX.match(tagPattern);
+    const insertIndex = match.index + match[0].length;
+
+    const finalBuild =
+      OLX.slice(0, insertIndex)
+      + `\n${' '.repeat(tabWidth * indentationLevel)}`
+      + `${questionBuild}`
+      + OLX.slice(insertIndex);
+
+    return finalBuild;
+  }
+
   buildMultiSelectProblem(problemType, widget, option) {
-    const question = this.addQuestion();
     const widgetObject = this.addMultiSelectAnswers(option);
     const demandhint = this.addHints();
     const problemObject = {
       problem: {
         [problemType]: {
-          ...question,
           [widget]: widgetObject,
         },
         ...demandhint,
       },
     };
-    return this.builder.build(problemObject);
+
+    const temporaryBuild = this.builder.build(problemObject);
```

Comment:
Nit: Perhaps change the name of the variable to something like `buildWithoutQuestion` or `olxWithoutQuestion`, so that it becomes apparent what this variable is.